src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -732,7 +732,7 @@ public StreamDirectory GetStreamDirectory()
                     throw new OrleansException("Failed to register " + typeof(TExtension).Name);
             }
 
-            IAddressable currentGrain = this.CurrentActivationData.GrainInstance;
+            IAddressable currentGrain = (runtimeContext.ActivationContext as SchedulingContext)?.Activation.GrainInstance;
             var currentTypedGrain = currentGrain.AsReference<TExtensionInterface>();
```

Comment:
you are using `?.` above, so I assume there is a chance for it to be null. If that's the case, then the AsReference call might throw a `NullReferenceException`